src/components/new-request/search.vue

```diff
@@ -459,10 +459,22 @@ export default class Search extends Mixins(CommonMixin, NrAffiliationMixin, Sear
     return (this.getEntityTypeCd === EntityTypes.CP)
   }
 
+  get isCreditUnion (): boolean {
+    return (this.getEntityTypeCd === EntityTypes.FI)
+  }
+
   get isSociety (): boolean {
     return (this.isSocietyEnabled() && this.getEntityTypeCd === EntityTypes.SO)
   }
 
+  /** Whether to show the name input field when trying to restore a historical company. */
+  get showRestoreNameInput (): boolean {
+    const isPromptNameInput = (this.isNamedCompany || this.isCooperative || this.isCreditUnion || this.isSelectedXproAndRestorable)
+    return (
+      this.isRestorable && isPromptNameInput && !this.isFederal
+    )
```

Comment:
The code is fine but is missing the detail that the comments provided, such as "why are we checking these specific variables?". That's what got me in this rabbit hole initially. And there's nothing wrong with the checks but it's not clear that they are different types of checks -- for example, not just entity type check, but also flow checks. Could you include the detailed comments please?

Reply:
In your original conversation, you described (and included a uxpin screenshot) what this check is, and that business logic needs to be described here or a future dev will not understand this code.

Reply:
Yes... I should leave more comments here, I also get confused when I first tryin to fix it.
I typed the following comment under `showRestoreNameInput`, not sure is it cleared enough and my poor grammar...
`// In the Uxpin, when company's type is one of coop, credit union, and XPro, it is in different work flow with regular named company.`
`// Thus we need to consider them seprately to determine whether we should have the name input field.   `

Reply:
That's very good, but future developers won't have access to the uxpin, so you'll have to describe the groups of companies that are included here.

1. named when radio button was used
2. specially include CP
3. specially include FI
4. include all xpros

Reply:
That looks great! Commit it and then check with Karim if this PR is OK to merge.